arangod/Cluster/Maintenance.cpp

```diff
@@ -796,35 +900,87 @@ arangodb::Result arangodb::maintenance::diffPlanLocal(
     }
   }
 
+  // Replicated Logs and States
   for (auto const& dbname : dirty) {
     using namespace arangodb::replication2;
-    auto lit = localLogs.find(dbname);
-    if (lit == std::end(localLogs)) {
-      continue;
-    }
-    auto pit = plan.find(dbname);
-    if (pit == std::end(plan)) {
+    if (!plan.contains(dbname) || !current.contains(dbname) ||
+        !localLogsByDatabase.contains(dbname)) {
```

Comment:
I think, whenever a database is dirty, it will have an entry in `plan` and `current`. This has nothing to do with what is actually written in the agency and might contain an empty object? But I am not sure. I will ask Max.

Reply:
Replicated Logs are worked on even if no current is available. hence current will eventually become available. Replicated States require current but also require replicated logs.